crates/noirc_evaluator/src/ssa/acir_gen.rs

```diff
@@ -18,12 +18,12 @@ use internal_var_cache::InternalVarCache;
 // Expose this to the crate as we need to apply range constraints when
 // converting the ABI(main parameters) to Noir types
 pub(crate) use constraints::range_constraint;
-mod memory_map;
-use memory_map::MemoryMap;
+mod acir_mem;
+use acir_mem::AcirMem;
 
 #[derive(Default)]
 pub struct Acir {
-    memory_map: MemoryMap,
+    memory_trace: AcirMem,
```

Comment:
`memory_trace` seems odd to me. What is it tracing? Can we call this just `memory` instead as it holds all the pseudo-memory in the program?